docs/proposals/14-reversed-cluster-vpn.md

```diff
@@ -0,0 +1,276 @@
+
+The OpenVPN server running in the shoot cluster is configured as follows:
+
+```
+mode server
+tls-server
+proto tcp4-server
+dev tun0
+
+[...]
+
+server 192.168.123.0 255.255.255.0
+
+push "route 10.243.0.0 255.255.128.0"
+push "route 10.243.128.0 255.255.128.0"
+
+duplicate-cn
+
+key "/srv/secrets/vpn-shoot/tls.key"
+cert "/srv/secrets/vpn-shoot/tls.crt"
+ca "/srv/secrets/vpn-shoot/ca.crt"
+dh "/srv/secrets/dh/dh2048.pem"
+
+tls-auth "/srv/secrets/tlsauth/vpn.tlsauth" 0
+push "route 10.242.0.0 255.255.0.0"
+```
+
+It is a TCP TLS server and configured to automatically assign IP addresses for OpenVPN clients (`server` directive). In addition, it pushes the shoot cluster node-, pod-, and service ranges to the clients running in the seed cluster (`push` directive). 
+
+**Note:** The network mesh spanned by OpenVPN uses the network range `192.168.123.0 - 192.168.123.255`. This network range cannot be used in either shoot-, or seed clusters. If it is used this might cause subtle problem due to network range overlaps. Unfortunately, this appears not to be well documented but this restriction exists since the very beginning. We should clean up this technical debt as part of the excercise.
+
+### Goals
+
+* We intend to supersede the current VPN solution with the solution outlined in this proposal.
+* We intend to remove the code for the konnectivity tunnel once this solution proposal has been validated.
+
+### Non Goals
+
+* The solution is not a low latency, or high throughput solution. As the kube-apiserver to shoot cluster traffic does not demand these properties we do not intend to invest in improvements.
+* We do not intend to provide continuous availability to the shoot-seed VPN connection. We expect the availability to be comparable to the existing solution.
+
+## Proposal
+
+The proposal is depicted in the following diagram:
+
+![alt text](assets/ReversedTunnelVPN.png "Reversed Tunnel VPN")
+
+We have added an OpenVPN server pod (`vpn-seed-server`) to each control plane. The OpenVPN client in the shoot cluster (`vpn-shoot-client`) connects to the OpenVPN server.
+
+The two containers `vpn-seed-server` and `vpn-shoot-client` are new containers and are not related to containers in the github.com/gardener/vpn project. We will create a new project github.com/gardener/vpn2 for these containers. With this solution we intend to supersede the containers from the github.com/gardener/vpn project.
+
+A service `vpn-seed-server` of type `ClusterIP` is created for each control plane in its namespace.
+
+The `vpn-shoot-client` pod connects to the correct `vpn-seed-server` service via the SNI passthrough proxy introduced with [SNI Passthrough proxy for kube-apiservers](https://github.com/gardener/gardener/blob/master/docs/proposals/08-shoot-apiserver-via-sni.md) port 8132. 
```

Comment:
```suggestion
The `vpn-shoot-client` pod connects to the correct `vpn-seed-server` service on port 8132 via the SNI passthrough proxy introduced with [SNI Passthrough proxy for kube-apiservers](https://github.com/gardener/gardener/blob/master/docs/proposals/08-shoot-apiserver-via-sni.md).
```

Reply:
The provided description is correct. The `vpn-seed-server` service is running on  a different port (1194). The public port is 8132.